### PR TITLE
Revision models.vision.unet, models.vision.segmentation

### PR DESCRIPTION
**Pull request by lijm1358**
## What does this PR do?

Related to #839

- [x] pl_bolts.models.vision.segmentation.cli_main
- [x] pl_bolts.models.vision.segmentation.SemSegment
- [x] pl_bolts.models.vision.unet.DoubleConv
- [x] pl_bolts.models.vision.unet.Down
- [x] pl_bolts.models.vision.unet.UNet
- [x] pl_bolts.models.vision.unet.Up

* update docstring, typing hints
* set bias=False of Conv2d in DoubleConv before batch normalization
* add test for unet components
## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙂

### CODE REVIEW

File: pl_bolts/models/vision/unet.py [hunk 1]
```diff
@@ -67,38 +67,35 @@ def forward(self, x):
...
-@under_review()
 class DoubleConv(nn.Module):
-    """[ Conv2d => BatchNorm (optional) => ReLU ] x 2."""
+    """[ Conv2d => BatchNorm => ReLU ] x 2."""
 
     def __init__(self, in_ch: int, out_ch: int):
         super().__init__()
         self.net = nn.Sequential(
-            nn.Conv2d(in_ch, out_ch, kernel_size=3, padding=1),
+            nn.Conv2d(in_ch, out_ch, kernel_size=3, padding=1, bias=False),
```
Comment (otaj) [inline pl_bolts/models/vision/unet.py:76]: Why are you explicitly removing bias?

**Review by otaj — APPROVED** [review]
Hi @lijm1358, it looks very good, thank you! I have left one question in the comment, plus I have another question - where is the number `250` as `ignore_index` for `cross_entropy` coming from? I assume it's from KITTI and if that's true, then it would be nice to have it also as an argument for the `SemSegment` class

File: pl_bolts/models/vision/unet.py (hunk 1 above)
Comment (lijm1358) [reply pl_bolts/models/vision/unet.py:76]: Bias of convolution before batch normalization can be removed because the beta term in batch normalization can replace the role of bias(https://discuss.pytorch.org/t/any-purpose-to-set-bias-false-in-densenet-torchvision/22067), and `nn.Conv2d` has default value of `bias=True`.

**Comment by lijm1358** [comment]
> Hi @lijm1358, it looks very good, thank you! I have left one question in the comment, plus I have another question - where is the number `250` as `ignore_index` for `cross_entropy` coming from? I assume it's from KITTI and if that's true, then it would be nice to have it also as an argument for the `SemSegment` class

Thank you for review, @otaj ! Yes, it's from KITTI datset. The labels useless for training are set to 250 in `datasets/kitti_dataset.py`. I'll add `ignore_index` as an argument for the `SemSegment`.

File: pl_bolts/models/vision/unet.py (hunk 1 above)
Comment (otaj) [reply pl_bolts/models/vision/unet.py:74]: Thank you for the pointer! It's a piece of information I somehow had stored very far in my head, but just didn't connect it with the proper reason why is it done here :) You're absolutely right :zap:
